src/identity/identity.service.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { Inject, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import authConfiguration, { AuthConfig } from '../config/auth.config';
+import { FunctionalityDisabledError, NotInDBError } from '../errors/errors';
+import { ConsoleLoggerService } from '../logger/console-logger.service';
+import { User } from '../users/user.entity';
+import { checkPassword, hashPassword } from '../utils/password';
+import { Identity } from './identity.entity';
+import { ProviderType } from './provider-type.enum';
+import { getIdentityFromUser } from './utils';
+
+@Injectable()
+export class IdentityService {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    @InjectRepository(Identity)
+    private identityRepository: Repository<Identity>,
+    @Inject(authConfiguration.KEY)
+    private authConfig: AuthConfig,
+  ) {
+    this.logger.setContext(IdentityService.name);
+  }
+
+  /**
+   * Check if the local registration is enabled
+   * @throws {FunctionalityDisabledError} user registration is disabled
+   */
+  checkIfRegistrationEnabled(): void {
+    if (!this.authConfig.email.enableRegister) {
+      this.logger.debug(
+        'User registration is disabled.',
+        'checkIfRegistrationEnabled',
+      );
+      throw new FunctionalityDisabledError('User registration is disabled.');
+    }
+  }
+
+  /**
+   * Check if the local login is enabled
+   * @throws {FunctionalityDisabledError} user login is disabled
+   */
+  checkIfLoginIsEnabled(): void {
```

Comment:
Should probably be called `localLogin` or `localAuth`, as it does not check if *any* login is enabled.